pkg/controller/account/byoc.go

```diff
@@ -59,6 +56,61 @@ func claimBYOCAccount(r *ReconcileAccount, reqLogger logr.Logger, currentAcctIns
 	return nil
 }
 
+// Checks whether or not the access keys need to be rotated based on state, and if so, rotates them
+func (r *ReconcileAccount) initializeCredentials(reqLogger logr.Logger, awsSetupClient, client awsclient.Client, currentAcctInstance *awsv1alpha1.Account, accountClaim *awsv1alpha1.AccountClaim, adminAccessArn string) (string, error) {
+
+	err := r.byocRotateAccessKeys(reqLogger, client, accountClaim)
+	if err != nil {
+		return "", err
+	}
+	roleID, err = createBYOCAdminAccessRole(reqLogger, awsSetupClient, client, adminAccessArn)
+	if err != nil {
+		return roleID, err
+	}
+
+	return roleID, err
+}
+
+func (r *ReconcileAccount) initializeNewBYOCAccount(reqLogger logr.Logger, currentAcctInstance *awsv1alpha1.Account, awsSetupClient awsclient.Client, adminAccessArn string) (string, error) {
+	client, accountClaim, err := r.getBYOCClient(currentAcctInstance)
+	if err != nil && accountClaim != nil {
```

Comment:
Should this be an `or`?

Reply:
I see what's happening. But I think we need to check if the error is nil, or if we can't find the accountclaim. Not sure what that logic should look like

Reply:
Good catch, thanks.  Addressed so that if there is an error, and the account claim is not nil, it calls `r.accountClaimBYOCError`, and then regardless, returns the error.